subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/server/http/HttpServerFixture.groovy

```diff
@@ -99,7 +105,7 @@ trait HttpServerFixture {
         }
 
         connector = new SocketConnector()
-        connector.port = 0
+        connector.port = enablePortAllocator ? FixedAvailablePortAllocator.instance.assignPort() : 0
```

Comment:
:x: `FixedAvailablePortAllocator` needs to release the port once assigned e.g. when server is shut down. If not then it stays reserved.

Reply:
:x: There are actually two more places in the class where we dynamically assign a port.

1. `private final Server server = new Server(0)` which internally already creates a `SocketConnector`. [line 35]
2. `connector = new SocketConnector(); connector.port = 0` [line 119]

Reply:
Overall I am not convinced this is the right approach.